test/llvmpasses/alloc-opt.jl

```diff
@@ -0,0 +1,204 @@
+# RUN: julia --startup-file=no %s | opt -load libjulia.so -AllocOpt -S - | FileCheck %s
+
+## WIP
+
+isz = sizeof(UInt) == 8 ? "i64" : "i32"
```

Comment:
While the integer size is easy to fix, its actually a pretty complex logic. The pass relies on details on object layout which not only depend on 32/64bit but also the architecture and even the OS. In particular, the pool offset depends on other members in the tls buffer including the size of `jl_jmp_buf`.

P.S. just notice that the `WIP` should be removed.....